workflow/src/main/java/bio/terra/tanagra/workflow/TextSearchUtils.java

```diff
@@ -0,0 +1,90 @@
+package bio.terra.tanagra.workflow;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.beam.sdk.transforms.Distinct;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.join.CoGbkResult;
+import org.apache.beam.sdk.transforms.join.CoGroupByKey;
+import org.apache.beam.sdk.transforms.join.KeyedPCollectionTuple;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TupleTag;
+import org.apache.beam.sdk.values.TypeDescriptors;
+
+public final class TextSearchUtils {
+  private TextSearchUtils() {}
+
+  /**
+   * Concatenate all search strings for a node into a single string. e.g. (123, "apple pie"), (123,
+   * "fruit dessert"), (123, "apple crumble") => (123, "apple pie , fruit dessert , apple crumble")
```

Comment:
Are there drawbacks to using "," as the delimiter here? Should we be escaping actual ","s in the input value and search string?

Reply:
Good point. I think we probably should be either escaping whatever the delimiter is in the search strings, or picking a delimiter that's not allowed in the UI search box. For AoU-CB, it looks like the following characters are not allowed: `~ @ [ ] | < > ( )`. I guess we should enforce that not-allowed list on the backend too if we're going to depend on it for indexing.

I'll plan to tackle this as follow-on work and perhaps we can discuss in stand-up whether the not-allowed search characters is something we'd be open to supporting generally.